### PR TITLE
gradle: add 8

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
###### Description of changes

Add Gradle 8 and make it the default (`gradle` is `gradle_8`, it was `gradle_7` before). See https://docs.gradle.org/8.0/release-notes.html

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by liff — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 216307` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>jabref</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>apksigcopier</li>
    <li>apksigner</li>
    <li>cie-middleware-linux</li>
    <li>diffoscope</li>
    <li>fastddsgen</li>
    <li>fdroidserver</li>
    <li>ghidra</li>
    <li>ma1sd</li>
    <li>scenic-view</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>apkleaks</li>
    <li>gradle (gradle_8)</li>
    <li>gscan2pdf</li>
    <li>jadx</li>
    <li>kotlin-language-server</li>
    <li>mindustry</li>
    <li>mindustry-server</li>
    <li>mindustry-wayland</li>
    <li>pdfchain</li>
    <li>pdftk</li>
    <li>purple-signald</li>
    <li>signald</li>
  </ul>
</details>

**Comment by lorenzleutgeb** [comment]
Failed packages should now build. I pinned them to Gradle 7.

**Comment by liff** [comment]
Result of `nixpkgs-review pr 216307` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>jabref</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>gscan2pdf</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>apkleaks</li>
    <li>diffoscope</li>
    <li>gradle (gradle_8)</li>
    <li>jadx</li>
    <li>kotlin-language-server</li>
    <li>mindustry</li>
    <li>mindustry-server</li>
    <li>mindustry-wayland</li>
    <li>pdfchain</li>
    <li>pdftk</li>
    <li>purple-signald</li>
    <li>signald</li>
  </ul>
</details>

**Comment by liff** [comment]
Not sure what’s up with gscan2pdf. Probably not related to Gradle?

**Comment by lorenzleutgeb** [comment]
Looks like `gscan2pdf` has some failing tests. However, I don't see how that relates to Gradle. @SuperSandro2000 please reconsider merging.

**Review by liff — APPROVED** [review]

**Comment by bhankas** [comment]
ping @SuperSandro2000 anything more remaining on this? #207308 is waiting for this as well, and at least one other package is currently not working on unstable because of the issue that I believe might be related.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -18037,7 +18037,8 @@ with pkgs;
   gradleGen = gradle-packages.gen;
   gradle_6 = callPackage gradle-packages.gradle_6 { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:18038]: As a next step, please try to remove gradle 6

**Pull request merged**